src/platform.js

```diff
@@ -44,16 +44,40 @@ export class Platform {
    * @return {boolean}
    */
   isSafari() {
-    return /Safari/i.test(this.win.navigator.userAgent) && !this.isChrome();
+    return /Safari/i.test(this.navigator.userAgent) && !this.isChrome();
   }
 
   /**
-   * Whether the current browser a Chrome browser.
+   * Whether the current browser is a Chrome browser.
    * @return {boolean}
    */
   isChrome() {
     // Also true for MS Edge :)
-    return /Chrome|CriOS/i.test(this.win.navigator.userAgent);
+    return /Chrome|CriOS/i.test(this.navigator.userAgent);
+  }
+
+  /**
+   * Whether the current browser is a Facebook webview.
+   * @return {boolean}
+   */
+  isFacebook() {
```

Comment:
I think I'd prefer this to go into your extension and keep this class simple. We really don't want stuff like this hard coded in the core framework, nor depend on this.

Also the method name is a bit misleading: FB using Chrome Custom Tabs would not be detected (but there the referrer is set).